### ISSUE TITLE
UserPrincipalName is case sensitive when adding members to a Microsoft 365 group

### ISSUE THREAD

**Issue by amortsell** [pnp/powershell#480]
Powershell version: 7.1.2
PnP version: 1.4.0

Problem: I was working on a migration project and used SharePoint migration Tool  to migrate data and permissions from SharePoint 2013 to SharePoint Online and created Group connected Team sites to which I migrated the existing SP 2013 Team sites. When the migration was complete I wanted to move the members from the SP Members group into the Microosft 365 group and ran the following code snippet to get the members and add them as members of the Microsoft 365 group.

    $groupMembers = @()

    for($i = $membersGroup.Users.count - 1; $i -ge 0; $i--){
        $user = $membersGroup.Users[$i]
        $nameParts = $item.LoginName.Split("{|}");
        if ($nameParts[0] -eq "i:0#.f") {  # Only do this for users, not for groups. Assume the LoginName is on the form 
            i0#.f|membership|<email address> for users.
            $email = $nameParts[2]  
            $AADUser = Get-PnPAzureADUser -Identity $email
            if ($AADUser -ne $null -and $AADUser.AccountEnabled -eq $True) {
                $groupMembers += $email
            }
        }
    }

    Set-PnPMicrosoft365Group -Identity $targetID -Members $groupMembers

This failed on the Set-PnPMicrosoft365Group command with the error message Set-PnPMicrosoft365Group: Value cannot be null. (Parameter 'source')
After quite a bit of debugging and testing we found that the same code worked in another tenant. Hence we started comparing the details between the two tenants and found that in the tenant where it worked the UserPrincipalName matched the Email from the SharePoint users exactly, but in the tenant where it didn't work the UserPrincipalName was capitalized like Firstname.Lastname@tenant.com and the email contained only lowercase letters firstname.lastname@tenant.com. When we replace the following line of code

    $groupMembers += $email

with

    $groupMembers += $AADUser.UserPrincipalName

the script started working.

I think it would be great if it was possible to add members to a group regardless of UserPrincipalName casing, so that firstname.lastname@tenant.com would work even if the UserPrincipalName is capitalized.

**Comment by erwinvanhunen** [comment]
I just checked what's happening behind the scenes here. We call https://graph.microsoft.com/v1.0/users/<email> address to resolve the user and find the id (which we need to add the user to the group). But regardless of what casing we used, it resolved...

**Comment by amortsell** [comment]
@erwinvanhunen Yes, we ran a few more tests yesterday and realized that UPN casing was probably not the cause of the problem because when we ran the script on other groups we received the same error after the code was updated to use the actual UPN as described above. The primary suspect currently is that we get the error when trying to add users to an empty group, but this seems to work at other tenants so I suspect it is some setting within this specific tenant that causes the problem. I was planning to close this issue today but haven't had time yet. Closing it now.
